docs/docs/examples-tutorials/recipes/testing-distributed-services-with-tyk-opentelemetry-tracetest.mdx

```diff
@@ -0,0 +1,378 @@
+        versions: ["Default"],
+      },
+    },
+  };
+
+  const res = await fetch("http://tyk-gateway:8080/tyk/keys/create", {
+    ...params,
+    method: "POST",
+    body: JSON.stringify(data),
+  });
+
+  const { key } = (await res.json()) as { key: string };
+
+  return key;
+};
+
+const definition: TestResource = {
+  type: "Test",
+  spec: {
+    id: "ZV1G3v2IR",
+    name: "Import Pokemon",
+    trigger: {
+      type: "http",
+      httpRequest: {
+        method: "POST",
+        url: "${var:ENDPOINT}/pokeshop/import",
+        body: '{"id": ${var:POKEMON_ID}}',
+        headers: [
+          {
+            key: "Content-Type",
+            value: "application/json",
+          },
+          {
+            key: "Authorization",
+            value: "${var:AUTHORIZATION}",
+          },
+        ],
+      },
+    },
+    specs: [
+      {
+        selector: 'span[tracetest.span.type="database"]',
+        name: "All Database Spans: Processing time is less than 100ms",
+        assertions: ["attr:tracetest.span.duration < 100ms"],
+      },
+      {
+        selector: 'span[tracetest.span.type="http"]',
+        name: "All HTTP Spans: Status  code is 200",
+        assertions: ["attr:http.status_code = 200"],
+      },
+      {
+        selector: 'span[tracetest.span.type="http" name="GET" http.method="GET" net.peer.name = "pokeapi.co"]',
+        name: "The request matches the pokemon Id",
+        assertions: ['attr:http.url   =  "https://pokeapi.co/api/v2/pokemon/${var:POKEMON_ID}"'],
+      },
+    ],
+  },
+};
+
+const main = async () => {
+  const tracetest = await Tracetest(TRACETEST_API_TOKEN);
+
+  const key = await setup();
+
+  const test = await tracetest.newTest(definition);
+  await tracetest.runTest(test, {
+    variables: [
+      {
+        key: "ENDPOINT",
+        value: POKESHOP_DEMO_URL.trim(),
+      },
+      {
+        key: "POKEMON_ID",
+        value: `${Math.floor(Math.random() * 100) + 1}`,
+      },
+      {
+        key: "AUTHORIZATION",
+        value: key,
+      },
+    ],
+  });
+  console.log(await tracetest.getSummary());
+};
+
+main();
+```
+
+The output from the `tracetest/index.ts` script will show the test results with links to the Tracetest App.
+
+```bash title=Terminal
+2024-02-20 13:11:06 
+2024-02-20 13:11:06 Successful: 1
+2024-02-20 13:11:06 Failed: 0
+2024-02-20 13:11:06 
+2024-02-20 13:11:06 [✔️ Import Pokemon] #11 - https://app.tracetest.io/organizations/ttorg_ced62e34638d965e/environments/ttenv_b42fa137465c6e04/test/ZV1G3v2IR/run/11
+2024-02-20 13:11:06 
+```
+
+### Tracetest App Results
+
+![Tracetest App Results](../img/tyk-results.png)
+
+### Running the example
+
+Spin up the deployment and test execution.
+
+```bash title=Terminal
+docker-compose up
+```
+
+This will trigger the docker compose setup and immediately run the [trace-based tests using the Tracetest Typescript integration](../../tools-and-integrations/typescript.mdx) as part of the `tracetest-client` service.
```

Comment:
```suggestion
This will trigger the Docker compose setup and immediately run the [trace-based tests using the Tracetest Typescript integration](../../tools-and-integrations/typescript.mdx) as part of the `tracetest-client` service.
```